Composer/packages/server/src/workers/merge_worker.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Worker, isMainThread, workerData, parentPort } from 'worker_threads';
+
+import { SchemaMerger } from '@microsoft/bf-dialog/lib/library/schemaMerger';
+
+import { Path } from '../utility/path';
+import { BotProject } from '../models/bot/botProject';
+
+if (!isMainThread) {
+  const realMerge = new SchemaMerger(
+    [workerData.manifestFile, '!**/imported/**', '!**/generated/**'],
+    Path.join(workerData.dataDir, 'schemas/sdk'),
+    Path.join(workerData.dataDir, 'dialogs/imported'),
+    false,
+    false,
+    console.log,
+    console.warn,
+    console.error
+  );
+
+  realMerge
+    .merge()
+    .then(() => {
+      process.exit(0);
+    })
+    .catch((err) => {
+      parentPort?.postMessage({ error: err });
+      process.exit(1);
+    });
+}
+
+export function startMerge(manifestFile: string, currentProject: BotProject) {
```

Comment:
Please name this exported function in a way so the reader on the caller end knows this is a worker thread doing something.
+ (It's not just starting the work, it's doing it)

Reply:
To add to this, the caller doesn't even need to know that this is a 'worker'. It can be named something like `mergeSchemas` and the caller can just await it.

Reply:
Handled